js/bcex.js

```diff
@@ -0,0 +1,399 @@
+        await this.loadMarkets ();
+        let request = {
+            'symbol': this.marketId(symbol)
+        }
+        if (typeof limit !== 'undefined')
+            request['limit'] = limit;
+        let market = this.market (symbol);
+        let response = await this.publicPostApiOrderMarketOrder(this.extend(request, params));
+        return this.parseTrades (response['data'], market, since, limit);
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        let response = await this.privatePostApiUserUserBalance (params)
+        let data = response['data'];
+        let keys = Object.keys(data);
+        let result = { }
+        for (let i = 0; i < keys.length; i++) {
+            let currentKey = keys[i];
+            let currentAmount = data[currentKey];
+            let split = currentKey.split('_')
+            let id = split[0];
+            let lockOrOver = split[1];
+            let currency = this.commonCurrencyCode(id);
+            if (!(currency in result)) {
+                let account = this.account ()
+                result[currency] = account;
+            }
+            if (lockOrOver == 'lock') {
+                result[currency]['used'] = parseFloat(currentAmount);
+            } else {
+                result[currency]['free'] = parseFloat(currentAmount);
+            }
+        }
+        keys = Object.keys(result)
+        for (let i = 0; i < keys.length; i++) {
+            let currentKey = keys[i];
+            let total = result[currentKey]['used'] + result[currentKey]['total']
+            result[currentKey]['total'] = total;
+        }
+        result['info'] = data
+        return this.parseBalance(result);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets();
+        let marketId = this.marketId(symbol);
+        let request = {
+            'symbol': marketId
+        };
+        let response = await this.privatePostApiOrderDepth(this.extend(request, params));
+        let data = response['data'];
+        let orderbook = this.parseOrderBook(data, data['date']);
+        return orderbook;
+    }
+
+    parseMyTrade (trade, market) {
+        let timestamp = trade['created'] * 1000;
+         return {
+            'id': trade['order_id'],
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': market.symbol,
+            'type': undefined,
+            'side': trade['side'],
+            'price': this.safeFloat (trade, 'price'),
+            'amount': this.safeFloat (trade, 'number'),
+            'order': undefined,
+            'fee': undefined,
+        };
+    }
+
+     parseMyTrades (trades, market = undefined, since = undefined, limit = undefined) {
+        let result = Object.values (trades || []).map (trade => this.parseMyTrade (trade, market))
```

Comment:
This line won't transpile properly, we have to unroll all maps to basic c-style for-loops. Otherwise this won't work with other languages.

```JavaScript
let result = [];
for (let i = 0; i < trades.length; i++) {
    result.push (parseMyTrade (trades[i], market)); // will edit for parseMyTrade → parseTrade
}
```